sklearn/experimental/_column_transformer.py

```diff
@@ -0,0 +1,427 @@
+    _fit_one_transformer, _fit_transform_one, _transform_one, _name_estimators)
+from ..utils import tosequence
+from ..utils.metaestimators import _BaseComposition
+from ..utils.validation import check_is_fitted
+
+
+__all__ = ['ColumnTransformer', 'make_column_transformer']
+
+
+_ERR_MSG_1DCOLUMN = ("1D data passed to a transformer that expects 2D data. "
+                     "Try to specify the column selection as a list of one "
+                     "item instead of a scalar.")
+
+
+class ColumnTransformer(_BaseComposition, TransformerMixin):
+    """Applies transformers to columns of an array or pandas DataFrame.
+
+    EXPERIMENTAL
+
+    This estimator allows different columns or column subsets of the input
+    to be transformed separately and the results combined into a single
+    feature space.
+    This is useful for heterogeneous or columnar data, to combine several
+    feature extraction mechanisms into a single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    Parameters
+    ----------
+    transformers : list of tuples
+        List of (name, transformer, column) tuples specifying the transformer
+        objects to be applied to subsets of the data.
+        Specifying the column as a scalar will pass a 1d array or pandas Series
+        to the transformer. Specifying the columns as a list, int array, slice
+        or boolean mask will pass a 2d array or DataFrame. Strings can
+        reference columns if the input is a DataFrame, integers are always
+        interpreted as the positional columns.
+        When passing a single column to a transformer that expects 2D input
+        data, the column should be specified a list of one element.
+
+    n_jobs : int, optional
+        Number of jobs to run in parallel (default 1).
+
+    transformer_weights : dict, optional
+        Multiplicative weights for features per transformer.
+        Keys are transformer names, values the weights.
+
+    Attributes
+    ----------
+    transformers_ : list
+        The collection of fitted transformers as tuples of
+        (name, fitted_transformer, column)
+
+    Examples
+    --------
+    >>> from sklearn.experimental import ColumnTransformer
+    >>> from sklearn.preprocessing import Normalizer
+    >>> union = ColumnTransformer(
+    ...     [("norm1", Normalizer(norm='l1'), [0, 1]),
+    ...      ("norm2", Normalizer(norm='l1'), slice(2, 4))])
+    >>> X = np.array([[0., 1., 2., 2.],
+    ...               [1., 1., 0., 1.]])
+    >>> union.fit_transform(X)    # doctest: +NORMALIZE_WHITESPACE
+    array([[ 0. ,  1. ,  0.5,  0.5],
+           [ 0.5,  0.5,  0. ,  1. ]])
+
+    """
+
+    def __init__(self, transformers, n_jobs=1, transformer_weights=None):
+        self.transformers = tosequence(transformers)
+        self.n_jobs = n_jobs
+        self.transformer_weights = transformer_weights
+        self._validate_transformers()
+
+    @property
+    def _transformers(self):
```

Comment:
Ah, this is for the implementation of ``get_params`` via ``_BaseComposition._get_params``. Maybe mention that?